tests/test_utils/test_helpers.py

```diff
@@ -51,3 +53,51 @@ def test_valuies_in_column(self, dummy_df, values, expected):
         assert result == expected, (
             "Returned value from values_in_column not as expected."
         )
+
+
+class TestTreeToList:
+    """Test for tree_to_list()"""
+
+    # Tests that a tree is correctly converted to a list.
+    # Tests that it raises a TypeError if the input is not a dictionary.
+
+    # Create a good input tree
+    def create_input_tree(self):
+        tree = {
+            "BERD": {
+                "01": {},
+                "02": {},
+            },
+            "PNP": {
+                "03": {},
+                "04": {"qa": {}},
+            },
+        }
+        return tree
+
+    # Create an input tree with a type error for negative test
+    def create_wrong_type_tree(self):
+
+        # The value must be an empty dictionary, not an empty list
+        bad_tree = {"BERD": []}
+        return bad_tree
+
+    # Create an expected dataframe for the test
+    def create_expected_list(self):
+        exp_output_list = [
+            'R:/2023/BERD', 'R:/2023/BERD/01', 'R:/2023/BERD/02', 'R:/2023/PNP',
+            'R:/2023/PNP/03', 'R:/2023/PNP/04', 'R:/2023/PNP/04/qa'
+        ]
+        return exp_output_list
+
+    def test_tree_to_list(self):
+        """Test for tree_to_list()"""
+        inp_tree = self.create_input_tree()
+        exp_output_list = self.create_expected_list()
+        bad_tree = self.create_wrong_type_tree()
+
+        result_list = tree_to_list(inp_tree, prefix="R:/2023")
+        assert result_list == exp_output_list
+
+        with pytest.raises(TypeError):
```

Comment:
Please add a matching message for the error raise here.

Reply:
done